metaflow/runner/subprocess_manager.py

```diff
@@ -29,6 +29,31 @@ def kill_process_and_descendants(pid, termination_timeout):
         pass
 
 
+async def async_send_signals(pids, signal):
+    pkill_processes = [
+        await asyncio.create_subprocess_exec("pkill", signal, "-P", str(pid))
+        for pid in pids
+    ]
+
+    for proc in pkill_processes:
+        await proc.wait()
+
+    kill_processes = [
+        await asyncio.create_subprocess_exec("kill", signal, str(pid)) for pid in pids
+    ]
+
+    for proc in kill_processes:
+        await proc.wait()
+
+
+async def async_kill_processes_and_descendants(pids, termination_timeout):
```

Comment:
can we simplify and kinda merge `async_send_signals` and `async_kill_processes_and_descendants`

also, the combined function could take one pid instead of multiple pids

and then we could do `async_kill` just like we have `kill` called on looping over these pids.

Reply:
I think this approach is slightly better, instead of doing it one by one, we can kill all of them at once. Although the impl was a bit silly and wasteful (too many processes), I've simplified it by calling each kill/pkill with an unpacked list of pids.

I've done the same for the sync version, this should be way faster for multiple managed subprocesses. Instead of waiting `n * termination_timeout` we can do only `termination_timeout` with the same effect.

Reply:
Actually, maybe scratch that. The sync `run_command` interface waits for the command to terminate before returning. Tbh we'd be fine by just calling kill on the current process, if we kept a reference to it somewhere. Looping over all of the commands is just going to end up in most of them failing, while still waiting 2 seconds for each.

Reply:
thanks, resolving..